syntax = "proto3";

package pars;

message settings
{
  string          dataset_filepath                = 1;
  repeated int32  dataset_size                    = 2;
                                                  
  repeated int32  seed_generation_stride          = 3;
  int32           seed_generation_iterations      = 4;
                                                  
  int32           particle_tracing_integrator     = 5;
  float           particle_tracing_step_size      = 6;
  bool            particle_tracing_load_balance   = 7;
                                                  
  int32           color_generation_mode           = 8;
  float           color_generation_free_parameter = 9;
                                                  
  repeated float  raytracing_camera_position      = 10;
  repeated float  raytracing_camera_forward       = 11;
  repeated float  raytracing_camera_up            = 12;
  repeated int32  raytracing_image_size           = 13;
  float           raytracing_streamline_radius    = 14;
  int32           raytracing_iterations           = 15;
}